musicApp/.Rproj.user/5ABFF95C/sources/per/t/C1CBF118-contents

```diff
@@ -0,0 +1,169 @@
+library(spotifyr)
+library(geniusr)
+library(dplyr)
+library(plyr)
+library(tidyr)
+library(jpeg)
+library(visNetwork)
+library(tm)
+
+Sys.setenv(GENIUS_API_TOKEN = 'yYDySPPgWepb-YB3ikWztEzGA1828BGpk9xyjZE91FR6fjfQF71zSmv6tYc3y5gt')
+Sys.setenv(SPOTIFY_CLIENT_ID = 'da1a67c279f04bc6a5996f6a7144e45c')
+Sys.setenv(SPOTIFY_CLIENT_SECRET = '2d4b1432f0e449a89806fd1d80409f36')
+
+
+####################################################################################################
+# function that returns list of nodes and edges of an artist's collaboration network based on a name
+getNetwork <- function(artist, progressBar = FALSE, weighted = FALSE)
+{
+  artistName = search_spotify(artist, limit = 1, type = 'artist')$name
+  artistId = search_spotify(artist, limit = 1, type = 'artist')$id
+
+  nodes = get_artist_audio_features(artistName, include_groups = c('album', 'single'))[c('artists', 'track_name')] %>%
+    unique() %>%
+    unnest(artists) %>%
+    select(name, id) %>%
+    count() %>%
+    filter(!id == artistId)
+
+
+  # here i make a matrix of edges (edgelist) by looping thtough each feature and seeing if they have mutual features with the selected artist:
+  edges = data.frame(from = character(), to = character())
+  for (i in 1:nrow(nodes))
+  {
+    if (progressBar) incProgress(amount = 1/nrow(nodes), detail = nodes$name[i])
+    collabId = nodes$id[i]
+    collabName = nodes$name[i]
+
+    tryCatch({
+      nodes2 = get_artist_audio_features(collabName, include_groups = c('album', 'single'))[c('artists', 'track_name')] %>%
+        unique() %>%
+        unnest(artists) %>%
+        select(id) %>%
+        filter(id != collabId & id != artistId) %>%
+        filter(id %in% nodes$id)                           # only select features that are in list of main artist features
+
+      if (nrow(nodes2) > 0)
+      {
+        edges = rbind(edges, data.frame(from = collabId, to = nodes2, stringsAsFactors = FALSE)) # add edges
+      }
+    },error=function(e){})
+  }
+
+  # sort each edge (so they are in the same order, and count duplicates to get weighted edgelist)
+  if(weighted)
+  {
+    edges = t(apply(edges,1,sort))
+    edges = plyr::count(edges)
+  }
+
+
+  network = list(nodes = nodes, edges = edges)
+
+  return(network)
+}
+
+
+####################################################################################################
+# function that returns a dataframe containing the urls of artist images based on a vector of ids
+getImages <- function(artistIds)
```

Comment:
Here you use artistIds as input. But how will the user know these IDs? I see an example of Ids is obtained in the previous function. You could return this to the user, or you could choose to ask the user to provide artist names instead of IDs (and then look for the corresponding Ids in the function, just like in the first function). This might be more user friendly.